docs/source/snapshot.rst

```diff
@@ -0,0 +1,104 @@
+Snapshot
+========
+
+:class:`~torchsnapshot.Snapshot` is the core API of TorchSnapshot. The class represents application state persisted in storage. A user can take a snapshot of an application via :meth:`Snapshot.take() <torchsnapshot.Snapshot.take>` (a.k.a saving a checkpoint), or restore the state of an application from a snapshot via :meth:`Snapshot.restore() <torchsnapshot.Snapshot.restore>` (a.k.a loading a checkpoint).
```

Comment:
There's a lot of references to "Application" but its not really defined, lets mention that this is used primarily for capturing state for training PyTorch models somewhere